### PR TITLE
feat: [CC-521] Add video frame adjustments for cover view

### PR DESCRIPTION
**Pull request by will-ltk**
Adds frame adjustments for the cover view when doing trim and cover.
Now, the video and cover view share the same adjustment logic.

### CODE REVIEW

File: Source/Filters/Photo/YPCoverImageView.swift [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+//
+//  Created by Will Saults on 1/15/24.
+//
+
+import UIKit
+import Photos
+
+public class YPCoverImageView: YPAdjustableView {
+
+    internal var coverImageView = UIImageView()
```
Comment (zeph-cohen10) [inline Source/Filters/Photo/YPCoverImageView.swift:13]: Does this need to be explicitly internal?

File: Source/Helpers/YPAdjustableView.swift [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    // Method to adjust the player view frame considering the asset's original size
+    private func adjustPlayerViewFrame(cropRect: CGRect, assetSize: CGSize) {
+        let assetRect = CGRect(origin: .zero, size: assetSize)
+        let fitFrame = calculateAspectFitFrame(assetSize: assetSize)
+        let targetFrame = calculateTargetFrame(cropRect: cropRect, assetRect: assetRect, fitFrame: fitFrame)
+        updateViewFrameAction?(targetFrame)
+    }
+
+    // Method to adjust the player view frame based on a target aspect ratio
+    private func adjustPlayerViewFrameForAspectRatio(cropRect: CGRect, aspectRatio: CGFloat, assetSize: CGSize) {
```
Comment (zeph-cohen10) [inline Source/Helpers/YPAdjustableView.swift:33]: It looks like most of this code isn't new just migrated out of the extension?

File: Source/Filters/Photo/YPCoverImageView.swift (hunk 1 above)
Comment (will-ltk) [reply Source/Filters/Photo/YPCoverImageView.swift:13]: Nope, that can come out. Thanks!

File: Source/Helpers/YPAdjustableView.swift (hunk 2 above)
Comment (will-ltk) [reply Source/Helpers/YPAdjustableView.swift:33]: It's new for the video player. I moved it here to support the cover image. Looks like I need to change this naming.

**Review by zeph-cohen10 — APPROVED** [review]

**Pull request merged**